modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java

```diff
@@ -172,9 +172,10 @@ private void grow(int capacityRequired, int bucketIndex) {
         }
 
         if (bucketIndex >= arrayBucket.length) {
+            int newLength = arrayBucket.length + (DEFAULT_ARRAY_BUCKET_SIZE > (bucketIndex - arrayBucket.length) ?
+                    DEFAULT_ARRAY_BUCKET_SIZE : (bucketIndex - arrayBucket.length)) + 1;
```

Comment:
Increasing array size by 1 will cause performance issues, when array access happens in an iteration, because each loop cause an Array.copy. So increment value should be a larger value like 100,500, DEFAULT_ARRAY_BUCKET_SIZE .